plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,500 @@
+## - Checks -  https://docs.sensu.io/sensu-go/latest/reference/checks
+## - Entities - https://docs.sensu.io/sensu-go/latest/reference/entities
+## - Metrics - https://docs.sensu.io/sensu-go/latest/reference/events#metrics
+##
+## Check specification
+## The check name is the name to give the Sensu check associated with the event
+## created.
+[outputs.sensu-go.check]
+  name = "telegraf"
+
+## Entity specification
+## Configure the entity name and namepsace, if necessary.
+##
+## NOTE: if the output plugin is configured to send events to a
+## backend_api_url and entity_name is not set, the value returned by
+## os.Hostname() will be used; if the output plugin is configured to send
+## events to an agent_api_url, entity_name and entity_namespace are not used.
+# [outputs.sensu-go.entity]
+#   name = "server-01"
+#   namespace = "default"
+
+## Metrics specification
+## Configure the tags for the metrics that are sent as part of the Sensu event
+# [outputs.sensu-go.tags]
+#   source = "telegraf"
+
+## Configure the handler(s) for processing the provided metrics
+# [outputs.sensu-go.metrics]
+#   handlers = ["influxdb","elasticsearch"]
+`
+
+// Description provides a description of the plugin
+func (s *Sensu) Description() string {
+	return "Send aggregate metrics to Sensu Monitor"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (s *Sensu) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Sensu) createClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: s.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (s *Sensu) Connect() error {
+	if s.Timeout.Duration == 0 {
+		s.Timeout.Duration = defaultClientTimeout
+	}
+
+	if len(s.ContentEncoding) != 0 {
+		validEncoding := []string{"identity", "gzip"}
+		if !choice.Contains(s.ContentEncoding, validEncoding) {
+			return fmt.Errorf("Unsupported content_encoding [%q] specified", s.ContentEncoding)
+		}
+	} else {
+		s.ContentEncoding = "identity"
+	}
+
+	if s.BackendApiUrl != nil && s.ApiKey == nil {
+		return fmt.Errorf("backend_api_url [%q] specified, but no API Key provided", *s.BackendApiUrl)
+	}
+
+	err := s.setEndpointUrl()
+	if err != nil {
+		return err
+	}
+
+	err = s.setEntity()
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.Timeout.Duration)
+	defer cancel()
+
+	client, err := s.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	s.client = client
+
+	return nil
+}
+
+func (s *Sensu) Close() error {
+	s.client.CloseIdleConnections()
+	return nil
+}
+
+func (s *Sensu) Write(metrics []telegraf.Metric) error {
+	var points []*OutputMetric
+	for _, metric := range metrics {
+		// Add tags from config to each metric point
+		var tagList []*OutputTag
```

Comment:
You might want to preallocate `tagList` by
```suggestion
		tagList := make([]*OutputTag, 0, len(s.Tags)+len(metric.TagList()))
```
to avoid reallocation.